### ISSUE TITLE
Review PR

### ISSUE THREAD

**Issue by gakonst** [naps62/pair-gpt.nvim#1]
1. Ask it "code review this branch against this other branch"
2. Gives you a `git add -p`-like interface which also shows an explanation of each patch
3. Then you `git commit -m <...>`

**Comment by naps62** [comment]
These sound like great suggestions, and not something I even thought of before.

The only potential caveat I see here is that this probably would only work for smaller diffs, due to the token limit of the API. but maybe that's what you were aiming at?

From the [openai help page](https://help.openai.com/en/articles/4936856-what-are-tokens-and-how-to-count-them):

> Depending on the [model](https://beta.openai.com/docs/engines/gpt-3) used, requests can use up to 4097 tokens shared between prompt and completion. If your prompt is 4000 tokens, your completion can be 97 tokens at most.

which seems to correspond to a hard limit of roughly 1000 chars (which would only leave room for a response of ~25 chars). Probably only usable for small bug fixes and not major features

**Comment by gakonst** [comment]
Ah I see. I think you can hack that around by asking it to remember the full response and give it to you back in chunks. You need to use more than 1 requests and perhaps tell it something like "give it to me in 10 parts", "now give me the second" etc.

**Comment by naps62** [comment]
true. I need to look into how that works. From the API docs I didn't find anything analogous to how the web chat works, where it clearly has the context of previous messages

Everything I've seen yesterday (such as [this](https://community.openai.com/t/getting-around-max-tokens/18103)) was suggesting to summarize the prompt, which maybe works for written text, but obviously not for code

I'd also bet these limits are short-term limitations only (due to both model limitations and server capacity)

**Comment by naps62** [comment]
~~Done~~

I'm senile. closed the wrong issue by mistake
